TiktokToOSCVRChat.py

```diff
@@ -0,0 +1,78 @@
+import os
+import asyncio
+import logging
+from aioudp import UDPServer
+from TikTokLive import TikTokLiveClient
+from TikTokLive.types.events import GiftEvent, CommentEvent
+import concurrent.futures
+import socket
+
+# Configuration options
+UDP_IP = "127.0.0.1"
+UDP_PORT = 9000
+USERNAME = os.getenv('@username')  # Get username from environment variables
+SLEEP_DURATION = 2  # Duration to sleep after sending a gift or boop
+RETRIES = 5  # Number of retry attempts
+
+def is_udp_server_running(ip, port):
+    # Check if the UDP server is running.
+    sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    try:
+        sock.bind((ip, port))
+    except socket.error as e:
+        return True
+    sock.close()
+    return False
+
+async def send_request(parameter, value):
+    # Send a request to the IP address with the current state of a parameter.
+    if is_udp_server_running(UDP_IP, UDP_PORT):
+        async with UDPServer(UDP_IP, UDP_PORT) as server:
+            await server.send_message(f"/avatar/parameters/{parameter}", value)
+    else:
+        logging.error(f"UDP server is not running on {UDP_IP}:{UDP_PORT}")
+
+async def handle_gift(event: GiftEvent):
+    # Handle gift events.
+    allowed_events = ["Rose", "Galaxy"]
+    event_name = event.gift.name
+    if event_name in allowed_events:
+        await asyncio.run_in_executor(None, send_request, event_name.lower(), 1)
+        await asyncio.sleep(SLEEP_DURATION)
+        await asyncio.run_in_executor(None, send_request, event_name.lower(), 0)
```

Comment:
This pattern repeats in lines `48-50` and seems rather common. I think it is best if we extract it to its own function:

```
async def run_asyncio_executor(f, event, sleep_secs = SLEEP_DURATION):
  await asyncio.run_in_executor(None, f, event, 1)
  await asyncio.sleep(sleep_secs)
  await asyncio.run_in_executor(None, f, event_name.lower(), 0)
```

```
async def handle_gift(event: GiftEvent):
    # Handle gift events.
    allowed_events = ["Rose", "Galaxy"]
    event_name = event.gift.name
    if event_name in allowed_events:
        await run_asyncio_executor(send_request, event_name.lower()) # Technically, this await can be removed
```